awx/ui/client/src/login/authenticationServices/authentication.service.js

```diff
@@ -49,38 +41,29 @@ export default
                 }
                 return $rootScope.userLoggedIn;
             },
-
-            getToken: function () {
-                if ($rootScope.token) {
-                    return $rootScope.token;
-                }
-
-                let token = $cookies.get('token');
-
-                return token ? token.replace(/"/g, '') : undefined;
-            },
-
             retrieveToken: function (username, password) {
-                return $http({
-                    method: 'POST',
-                    url: GetBasePath('authtoken'),
-                    data: {
-                        "username": username,
-                        "password": password
-                    },
-                    headers: {
-                        'Cache-Control': 'no-store',
-                        'Pragma': 'no-cache'
-                    }
+                var getCSRFToken = $http({
+                    method: 'GET',
+                    url: `/api/login/`
+                });
+
+                return getCSRFToken.then(function({data}) {
+                    var csrfmiddlewaretoken = /name='csrfmiddlewaretoken' value='([0-9a-zA-Z]+)' \//.exec(data)[1];
+                    // TODO: data needs to be encoded
+                    return $http({
+                        method: 'POST',
+                        url: `/api/login/`,
+                        data: `username=${username}&password=${password}&csrfmiddlewaretoken=${csrfmiddlewaretoken}&next=%2fapi%2f`,
+                        headers: {
+                            'Content-Type': 'application/x-www-form-urlencoded'
+                        }
+                    });
                 });
             },
             deleteToken: function () {
                 return $http({
-                    method: 'DELETE',
-                    url: GetBasePath('authtoken'),
-                    headers: {
-                        'Authorization': 'Token ' + this.getToken()
-                    }
+                    method: 'GET',
```

Comment:
Should this be `POST` instead for logout? Not required, but there's significant discussion as to whether logout should be allowed via `GET`: https://code.djangoproject.com/ticket/15619